code/game/antagonist/antagonist_create.dm

```diff
@@ -99,11 +99,14 @@
 		player.current << "<span class='notice'>[leader_welcome_text]</span>"
 	else
 		player.current << "<span class='notice'>[welcome_text]</span>"
+	if (config.objectives_disabled)
+		player.current << "<span class='notice'>[antag_text]</span>"
 
 	if((flags & ANTAG_HAS_NUKE) && !spawned_nuke)
 		create_nuke()
 
-	show_objectives(player)
+	if (!config.objectives_disabled)
```

Comment:
The purpose of not checking config here was for people given objectives by admin via the interface, since making it depend on the config resulted in being unable to give people objectives without enabling it for all subsequent antagonists.